### PR TITLE
Clears empty rooms

### PR DESCRIPTION
**Pull request by Arkaeriit**
When a user left a room where they were alone in, the room was lest open.
This PR fixes that and ensure that there is no empty rooms.

### CODE REVIEW

File: devchat.go [hunk 1]
```diff
@@ -279,6 +279,16 @@ func newUser(s ssh.Session) *user {
 	return u
 }
 
+// Deletes any empty room. Checks all rooms so that it works even when the
+// user left.
+func cleanupRooms() {
+	for _, r := range rooms {
```
Comment (quackduck) [inline devchat.go:285]: We only need to check the room this user left, so a loop through all rooms is not necessary. I would have this function take in a room argument to check.

File: devchat.go [hunk 2]
```diff
@@ -289,6 +299,7 @@ func (u *user) close(msg string) {
 			u.room.broadcast(devbot, u.name+" stayed on for "+printPrettyDuration(time.Since(u.joinTime)))
 		}
 	})
+	cleanupRooms()
```
Comment (quackduck) [inline devchat.go:302]: This should be moved into the `closeOnce.Do` function

**Comment by quackduck** [comment]
Thank you! I _was_ wondering what had been causing the empty rooms lol

**Comment by Arkaeriit** [comment]
I fixed the problems you raised.

File: devchat.go [hunk 3]
```diff
@@ -279,6 +279,14 @@ func newUser(s ssh.Session) *user {
 	return u
 }
 
+// Deletes any empty room. Checks all rooms so that it works even when the
+// user left.
```
Comment (quackduck) [inline devchat.go:283]:
```suggestion
// cleanupRoom deletes a room if it's empty and isn't the main room
```

**Comment by quackduck** [comment]
Perfect, thank you so much!

**Pull request merged**